### PR TITLE
Enable Pauseless GC on macOS

### PR DESCRIPTION
**Pull request by RSalman**
Enable Concurrent Scavenger `-Xgc:concurrentScavenge` for macOS builds

Signed-off-by: Salman Rana <salman.rana@ibm.com>

### CODE REVIEW

**Comment by RSalman** [comment]
Depends on: https://github.com/eclipse/omr/pull/4546

**Comment by DanHeidinga** [comment]
This will need a doc update as well as I believe we've previously updated the docs to say CS wasn't supported on osx

File: runtime/gc_glue_java/configure_includes/configure_osx.mk [hunk 1]
```diff
@@ -32,7 +32,8 @@ CONFIGURE_ARGS += \
 	--enable-OMR_GC_IDLE_HEAP_MANAGER \
 	--enable-OMR_GC_TLH_PREFETCH_FTA \
 	--enable-OMR_PORT_CAN_RESERVE_SPECIFIC_ADDRESS \
-	--enable-OMR_PORT_NUMA_SUPPORT
+	--enable-OMR_PORT_NUMA_SUPPORT \
```
Comment (amicic) [inline runtime/gc_glue_java/configure_includes/configure_osx.mk]: hm, why enabling NUMA?

**Comment by amicic** [comment]
@DanHeidinga and I eventually agreed that no doc changes are required, since lack of OSX in fact was not documented.

**Comment by amicic** [comment]
We do however need to enable CS testing for OSX, after this is merged.

**Comment by pshipton** [comment]
> @DanHeidinga and I eventually agreed that no doc changes are required, since lack of OSX in fact was not documented.

I'd expect to have the additional of Mac in the release notes https://eclipse.github.io/openj9-docs/version0.18/ even if https://eclipse.github.io/openj9-docs/xgc/#concurrentscavenge doesn't need to be updated.

**Comment by pshipton** [comment]
and actually, https://eclipse.github.io/openj9-docs/xgc/#concurrentscavenge does need to be updated to include Mac in the platform list.

**Comment by pshipton** [comment]
Here is an example release note https://eclipse.github.io/openj9-docs/version0.14/#improved-support-for-pause-less-garbage-collection

The platform list also needs to be updated here https://eclipse.github.io/openj9-docs/gc/

**Comment by RSalman** [comment]
> We do however need to enable CS testing for OSX, after this is merged.

PR open to enabling tests: https://github.com/AdoptOpenJDK/openjdk-tests/pull/1443

**Comment by amicic** [comment]
Created doc change request: https://github.com/eclipse/openj9-docs/issues/433

**Comment by pshipton** [comment]
@amicic it's not enough to create a doc issue any more, we need to create a Pull Request with the doc changes as well. The docs team no longer has time to make doc changes and will only review them. Any questions, see https://github.com/eclipse/openj9-docs/issues/408

**Pull request merged**